src/lib/storage/local/local-data.js

```diff
@@ -15,10 +16,28 @@ const Path = require('path');
    */
    constructor(path) {
     this.path = path;
+    let dbFile;
+
     try {
-      this.data = JSON.parse(fs.readFileSync(this.path, 'utf8'));
-    } catch(_) {
-      this.data = {list: []};
+      dbFile = fs.readFileSync(this.path, 'utf8');
+    } catch (err) {
+      if (err.code === 'ENOENT') { // Only recreate if file not found to prevent data loss
+        this.data = {list: []};
+      } else {
+        logger.logger.error(
```

Comment:
Tests are failing because this change
```
2) basic system test "before all" hook:
     Error: Package database file corrupted (invalid JSON), please fix it manually.
File Path: /home/ubuntu/verdaccio/test/unit/partials/store/test-storage/.sinopia-db.json
```